### DISCUSSION TITLE
FEATURE REQUEST: Multiple nullstr values in read_csv_auto

### DISCUSSION THREAD

**Discussion by daplaci** [duckdb/duckdb#10800] [Ideas]
Hello,

Would be possible to have an array passed to nullstr in read_csv_auto, to handle cases when multiple values should be considered NULL (e.g. "NULL" and "")?

**Comment by pdet** [comment]
Hey @daplaci, there is nothing from a technical perspective that makes that not possible. I guess it wouldn't even be a breaking change, since these types can be resolved during binding.

It would, however, have a performance impact, since we would need to add verification steps for each value on the csv file.

Are you aware of other systems that implement multiple options for null strings for CSV reading? Can you also share more about the use-case where systems write csv files with multiple null representations?

**Reply by cmdlineluser** [reply]
Both Pandas and Polars accept str, list[str] and then a mapping for per-column values:

> `null_values`
>     Values to interpret as null values. You can provide a:
> 
- `str`: All values equal to this string will be null.
- `List[str]`: All values equal to any string in this list will be null.
- `Dict[str, str]`: A dictionary that maps column name to a null value string.

(Pandas calls its `na_values`)

**Reply by pdet** [reply]
Alright, it's fair to add support to that then.

**Reply by Tishj** [reply]
We currently mock this option in `duckdb.read_csv(..)` in Python, only allowing a single value:
```c++
	if (!py::none().is(na_values)) {
		if (!py::isinstance<py::str>(na_values)) {
			throw InvalidInputException("read_csv only accepts 'na_values' as a string");
		}
		bind_parameters["nullstr"] = Value(py::str(na_values));
	}
```

**Reply by arthurgailes** [reply]
In readr::read_csv, the default NA string is c("", "NA"). Base R's read.csv and data.table::fread also allow multiple null strings.

**Reply by ericmanning** [reply]
Arrow allows for a list of null strings:
https://arrow.apache.org/docs/python/generated/pyarrow.csv.ConvertOptions.html

Is this feature in the queue?

**Reply by gyansinha** [reply]
Has this feature been implemented yet?

**Reply by pdet** [reply]
Yes, nullstr can take a `VARCHAR[]` now.

**Reply by gyansinha** [reply]
Thanks. Sorry, I didn't see an example or reference in the docs, how would this work in the COPY command, with say 2 values for null "NA" and "".
